### PR TITLE
Removing material from checkbox component

### LINKED ISSUE #1187: Remove Material UI dependencies from Checkbox

### PR DESCRIPTION
**Pull request by Jialecl**
Closes #1187

### CODE REVIEW

**Review by raquelarrojo — CHANGES_REQUESTED** [review]
- When the checkbox is checked, if we make hover on the label, the checkbox should not be hovered. We cold add a test of this in Chromatic.
- When the checkbox is checked/unchecked, it should not be focused.

File: lib/src/checkbox/Checkbox.tsx [hunk 1]
```diff
@@ -77,37 +81,44 @@ const DxcCheckbox = ({
...
         margin={margin}
         size={size}
         backgroundType={backgroundType}
-        isLabelHovered={isLabelHovered}
+        checked={checked ?? innerChecked}
       >
         {label && labelPosition === "before" && labelComponent}
-        <Checkbox
+        <input
+          type="checkbox"
```
[the remark this answers is not in the document]
Comment (raquelarrojo) [reply lib/src/checkbox/Checkbox.tsx:93]: Already done.

**Comment by Jialecl** [comment]
> * When the checkbox is checked, if we make hover on the label, the checkbox should not be hovered. We cold add a test of this in Chromatic.

Since the unchecked checkbox was reacting to the label being hovered, we pondered if the checked checkbox should also change state and it was decided to add it.

File: lib/src/checkbox/Checkbox.tsx [hunk 2]
```diff
@@ -77,37 +81,44 @@ const DxcCheckbox = ({
     <ThemeProvider theme={colorsTheme.checkbox}>
```
[the remark this answers is not in the document]
Comment (raquelarrojo) [reply lib/src/checkbox/Checkbox.tsx:80]: This `labelPosition` check is still done twice: in `labelComponent` and in the return. Please keep it only in the return.

**Review by raquelarrojo — CHANGES_REQUESTED** [review]
There should not be any visual changes with the new version without material. We should fix this change:

image

File: lib/src/checkbox/Checkbox.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (Jialecl) [reply lib/src/checkbox/Checkbox.tsx:80]: The one that is done inside `labelComponent` is to put optional in the correct position. And the one that is done where the `labelComponent` is rendered is to position the `labelComponent` itself.

**Pull request closed without merging**